controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -294,7 +293,7 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 		if node == nil {
 			continue
 		}
-		if noderefutil.IsNodeReady(node) {
+		if node.Spec.ProviderID != "" {
```

Comment:
I'm not sure this is actually a correct way to determine Node readiness, but might be good enough until we have health checking in place.

Reply:
while reviewing I stumbled upon the same question, after reviewing the `IsNodeReady` function, it made sense mostly because we want the ProviderID to be set, but we shouldn't wait for the node condition to be "Ready" (because it can't be there yet)

Reply:
@vincepri Indeed, I think once we have a better definition of "health" for a control plane member, we'll be in better shape here to know that an actual instance is "ready" vs the node condition.